blocks/gmo-program-list/gmo-program-list.js

```diff
@@ -394,36 +401,58 @@ function repaginate(dropdown) {
     decorate(block, currentNumberPerPage, '', false, false);
 }
 
+/**
+ * Limits the rate at which a function can be executed by ensuring it is only called after a specified delay since the last invocation.
+ * @param {Function} func - The function to debounce.
+ * @param {number} wait - The delay in milliseconds.
+ * @returns {Function} - The debounced function.
+ */
+function debounce(func, wait) {
+    let timeout;
+    return function(...args) {
+        const later = () => {
+            clearTimeout(timeout);
+            func(...args);
+        };
+        clearTimeout(timeout);
+        timeout = setTimeout(later, wait);
+    };
+}
+
 function nextPage(nextBtn) {
-    if (currentPageInfo.hasNextPage) {
-      //Calculate Next Page
-      currentPage++;
-      const block = document.querySelector('.gmo-program-list.block');
-      decorate( block, currentNumberPerPage, currentPageInfo.nextCursor, false, true,currentGraphqlFilter);
-      if (!(nextBtn.classList.contains('active'))) {
-          return;
-      }
-      const prevBtn = document.querySelector('.footer-pagination-button.prev');
-      prevBtn.classList.add('active');
+    if (currentPage < totalPages) {
+        currentPage++;
+        const block = document.querySelector('.gmo-program-list.block');
+        decorate(block, currentNumberPerPage, currentPageInfo.nextCursor, false, true, currentGraphqlFilter);
+
+        const prevBtn = document.querySelector('.footer-pagination-button.prev');
+        prevBtn.classList.add('active');
+
+        if (currentPage === totalPages) {
+            nextBtn.classList.remove('active');
+        } else {
+            nextBtn.classList.add('active');
+        }
     }
 }
 
+
 function prevPage(prevBtn) {
-    if (currentPageInfo.hasPreviousPage) {
-      currentPage--;
-      const block = document.querySelector('.gmo-program-list.block');
-      const currentCursor = currentPageInfo.currentCursor;
-      //Calculate cursor for previous page
-      const indexCursor = cursorArray.indexOf(currentCursor) - currentNumberPerPage;
-      decorate(block, currentNumberPerPage, cursorArray[indexCursor], true, false,currentGraphqlFilter);
-      if (!(prevBtn.classList.contains('active'))) {
-          return;
-      }
-      const nextBtn = document.querySelector('.footer-pagination-button.next');
-      const currentPageBtn = document.querySelector('#current-page');
-      const currentPageValue = parseInt(currentPageBtn.dataset.pagenumber);
-      const targetPage = (currentPageValue - 1);
-      nextBtn.classList.add('active');
+    if (currentPage > 1) {
+        currentPage--;
+        const block = document.querySelector('.gmo-program-list.block');
+        const indexCursor = cursorArray.indexOf(currentPageInfo.currentCursor) - currentNumberPerPage;
+        const newCursor = indexCursor >= 0 ? cursorArray[indexCursor] : '';
```

Comment:
we can use `startCursor` here 
![image](https://github.com/hlxsites/adobe-gmo/assets/61603050/1107d47b-77aa-4350-b851-be39ee3cd361)